### PR TITLE
BuildResidentialHPXML measure: Expose emissions arguments for fossil fuels

### LINKED ISSUE #964: BuildResidentialHPXML measure: Expose emissions arguments for fossil fuels
We'd want to try to limit how many additional arguments we add. Consider a comma-separated list argument for fuel type. Update: this wouldn't work since each element of the comma-separated list is a scenario.

This would support ResStock-HPXML.

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Closes #964.

Add optional comma-separated arguments to the `BuildResidentialHPXML` measure:
* `emissions_fossil_fuel_units` 
* `emissions_xxx_values` (for `natural_gas`, `propane`, `fuel_oil`, `coal`, `wood`, `wood_pellets`)

This supports ResStock-HPXML.

## Checklist

Not all may apply:

- [ ] EPvalidator.xml has been updated
- [ ] Tests (and test files) have been updated
- [ ] Documentation has been updated
- [ ] Changelog has been updated
- [ ] `openstudio tasks.rb update_measures` has been run
- [ ] No unexpected regression test changes on CI (checked comparison artifacts)

### CODE REVIEW

File: BuildResidentialHPXML/measure.rb [hunk 1]
```diff
@@ -3101,13 +3126,50 @@ def argument_errors(args)
                                   args[:emissions_electricity_units].is_initialized,
                                   args[:emissions_electricity_values_or_filepaths].is_initialized]
     error = (emissions_args_initialized.uniq.size != 1)
-    errors << 'Did not specify either no emissions arguments or all emissions arguments.' if error
+    errors << 'Did not specify either no required emissions arguments or all required emissions arguments.' if error
```
[the remark this answers is not in the document]
Comment (joseph-robertson) [reply BuildResidentialHPXML/measure.rb:3129]: Yeah, I don't know. To me that sounds like the emissions arguments are required. I'm trying to say that it's "all or none".

[the remark this answers is not in the document]
Comment (joseph-robertson) [reply BuildResidentialHPXML/measure.rb:3129]: Although, I suppose you wouldn't see a message if you had none specified. Fine.

**Review by shorowit — COMMENTED** [review]
The code overall looks really good now.

But I have one question that I can't tell from reading the code. If you have a home that just uses electricity and natural gas, does the measure allow you to not enter values for propane, fuel oil, etc?

**Comment by joseph-robertson** [comment]
Yes, it should. If you enter values for a fossil fuel, it requires you to enter fossil fuel units. But shouldn't require that you enter all fossil fuel values.

**Review by shorowit — APPROVED** [review]

**Comment by joseph-robertson** [comment]
Ugh, maybe you're right. I think I need to re-write a bit.

**Pull request merged**